apps/climatemappedafrica/src/lib/data/blockify/hero.js

```diff
@@ -18,7 +15,7 @@ export default async function hero(block, _api, _context, { hurumap }) {
     rootGeography: { center, code, hasData: pinRootGeography, zoom },
     profile: hurumapProfile,
   } = hurumap ?? { rootGeography: {} };
-  const { geometries } = await fetchProfileGeography(code.toLowerCase(), {
+  const { geometries } = await fetchCachedProfileGeography(code.toLowerCase(), {
```

Comment:
Looks like this `fetchCachedProfileGeography` call has to be made for each `code`, twice if/when we're doing comparison (unlike the profile call which is the same for the whole app).
1. Doesn't this mess up the caching plan?
2. Isn't there a way of fetching for many codes in one call?

Reply:
@kilemensi There are around 815 geographies, but this looks to increase as some countries currently do not have regions. The only page we're building is `explore/af`. I tried building for all 55 countries, but the API became unresponsive due to the many requests.

Reply:
@kilemensi We may also have to rethink how to optimize it further, for example fetching data for [`ke`](https://climatemapped-africa.hurumap.org/api/v1/all_details/profile/1/geography/KE/?version=Climate) returns a response that is 2.6MB. In most countries I have tested, the response size is more than 2MB. This also extends to regions which also have ~2MB of response data each.

Reply:
Lets create /explore, /explore/af and 5 - 10 core HURUmap countries @kelvinkipruto; the rest will be created on demand.

If the initial page rendered for a non-built page is good enough all good, otherwise, we may have to change our [`fallback`](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-static-props#when-does-getstaticprops-run) policy to `blocking`.

>We may also have to rethink how to optimize it further, for example fetching data for [ke](https://climatemapped-africa.hurumap.org/api/v1/all_details/profile/1/geography/KE/?version=Climate) returns a response that is 2.6MB...
>
BE optimisation will be done down the line. On FE, we don't need to pass on the whole 2.6MB, since it's a JSON, you can check and return only the props/attributes we need.